include/envoy/stats/stats_macros.h

```diff
@@ -28,15 +28,18 @@ namespace Envoy {
 
 #define GENERATE_COUNTER_STRUCT(NAME) Stats::Counter& NAME##_;
 #define GENERATE_GAUGE_STRUCT(NAME) Stats::Gauge& NAME##_;
-#define GENERATE_TIMER_STRUCT(NAME) Stats::Timer& NAME##_;
+#define GENERATE_TIMER_STRUCT(NAME) Stats::Histogram& NAME##_;
```

Comment:
Can we get rid of this distinction in this file and just switch to pure histogram?

Reply:
I think this depends on if we decide to treat them differently from the user's perspective as discussed above.